bitcoin/src/blockdata/fee_rate.rs

```diff
@@ -112,10 +168,85 @@ impl Mul<Weight> for FeeRate {
 impl Div<Weight> for Amount {
     type Output = FeeRate;
 
-    fn div(self, rhs: Weight) -> Self::Output { FeeRate(self.to_sat() * 1000 / rhs.to_wu()) }
+    fn div(self, rhs: Weight) -> Self::Output {
+        FeeRate { per_kwu: self * 1000 / rhs.to_wu() }
+    }
 }
 
-crate::parse::impl_parse_str_from_int_infallible!(FeeRate, u64, from_sat_per_kwu);
+#[cfg(feature = "serde")]
+pub mod serde {
+    #![allow(missing_docs)]
+    //! A module for serde-serializing fee rates in various units.
+
+    use serde::de::{Deserialize, Deserializer, Error};
+    use serde::ser::Serializer;
+
+    use super::{Amount, FeeRate};
+    use crate::amount::Denomination;
+
+    pub mod btc_per_kvb {
+        //! Serialize fee rates as bitcoin per kilo-virtual bytes.
+        use super::*;
+
+        pub fn serialize<S: Serializer>(r: &FeeRate, s: S) -> Result<S::Ok, S::Error> {
+            s.serialize_f64((r.to_per_vb_ceil() * 1000).to_btc())
```

Comment:
Please comment on the use of ceil instead of floor here. Not entirely sure what the most sensible alternative is..

Reply:
IMO we should have `ciel`/`floor` in the module name.

Reply:
When creating a feerate I guess you want `ceil` and when checking you want `floor` ... I don't know if we really need both but definitely agreed we should signal which one we're using so that it's more obvious that users are making a choice.

Reply:
In the type I created, the base unit was sat per kvb so it didn't have this problem :sweat_smile:  I think the vast majority of usages will be handling vbyte anyway so the ceil/floor wouldn't even be a problem.

For that reason, I'm somehow inclined to want a "not explicitly ceil/floor" version for convenience (like, all the stuff core communicates in the RPC API is expressed in vb multiples so it will always be multiples of 4 in this type.

Also, why do you want floor for checking? If you check yours (which supposedly is neither floored not ceiled because it's native) with one that you receive, wouldn't you want it ceiled too? My reasoning was with flooring you risk not meeting thresholds, with ceiling you risk paying a sat too much worst case..

Reply:
When checking, if you use ceil you'll overestimate the fee rate (and therefore think the fee is high enough when it's not); with floor you'll underestimate (and maybe pay a sat more than you needed to).

IOW when checking, I think the comparison is reversed versus creating transactions.